Calibration/HcalCalibAlgos/macros/CalibCorr.C

```diff
@@ -0,0 +1,173 @@
+#include <algorithm>
+#include <map>
+#include <vector>
+#include <string>
+#include <iomanip>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+class CalibCorr {
+public :
+  CalibCorr(const std::string& infile, bool debug=false);
+  ~CalibCorr() {}
+
+  float getCorr(int run, unsigned int id);
+private:
+  void                     readCorr(const std::string& infile);
+  std::vector<std::string> splitString(const std::string&);
+  unsigned int getDetIdHE(int ieta, int iphi, int depth);
+  unsigned int getDetId(int subdet, int ieta, int iphi, int depth);
+  unsigned int correctDetId(unsigned int detId);
+
+  static const unsigned int nmax_=10;
+  bool                      debug_;
```

Comment:
I suggest to make this const as this is known already at construction time.

Reply:
done